building_gazebo_plugins/src/thumbnail_generator.cpp

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <string>
+#include <functional>
+#include <ignition/transport/Node.hh>
+#include <opencv2/opencv.hpp>
+
+#include "gazebo/msgs/msgs.hh"
+#include "gazebo/transport/transport.hh"
+#include "gazebo/sensors/SensorsIface.hh"
+#include "gazebo/common/Plugin.hh"
+#include "gazebo/rendering/rendering.hh"
+#include "gazebo/util/system.hh"
+
+#define IMG_SIZE 4000
+#define CAM_HFOV 0.08
+#define CAM_HEIGHT 200
+
+using namespace gazebo;
+
+class GZ_PLUGIN_VISIBLE ThumbnailGenerator : public SystemPlugin
+{
+
+public:
+  /////////////////////////////////////////////
+  virtual ~ThumbnailGenerator()
+  {
+    rendering::fini();
+  }
+
+  /////////////////////////////////////////////
+  void Load(int _argc, char** _argv)
+  {
+    if (_argc != 6)
```

Comment:
writing an arg parser with proper flagging like `--model` and `--output` might be a cleaner way to handle more arguments in the future.